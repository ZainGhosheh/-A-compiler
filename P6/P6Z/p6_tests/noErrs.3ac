[BEGIN GLOBALS]
func
[END GLOBALS]
[BEGIN <init> LOCALS]
[END <init> LOCALS]
fun_<init>: enter <init>
lbl_0:      leave <init>
[BEGIN func LOCALS]
a (local var of 8 bytes)
b (local var of 8 bytes)
c (local var of 8 bytes)
[END func LOCALS]
fun_func:   enter func
            [c] := [c] ADD64 1
lbl_1:      leave func

