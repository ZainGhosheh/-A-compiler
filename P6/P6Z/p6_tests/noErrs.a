callee : (arg1:int, arg2:int) -> int {
  return arg1 + arg2;
}

void_callee : (arg:bool) -> void {
  return;
}

main : () -> void {
  a:int = callee(3, 4);
  callee(5, 6);
  void_callee(true);
}