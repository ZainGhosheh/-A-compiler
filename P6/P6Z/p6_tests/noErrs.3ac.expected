[BEGIN GLOBALS]
func
[END GLOBALS]
[BEGIN <init> LOCALS]
[END <init> LOCALS]
fun_<init>: enter <init>
lbl_0:      leave <init>
[BEGIN func LOCALS]
a (local var of 8 bytes)
b (local var of 8 bytes)
tmp0 (tmp var of 8 bytes)
tmp1 (tmp var of 8 bytes)
[END func LOCALS]
fun_func:   enter func
            [tmp0] := [a] ADD64 4
            [a] := [tmp0]
            [tmp1] := [a] ADD64 3
            [b] := [tmp1]
            [a] := [a] ADD64 1
lbl_1:      leave func